bin/https.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <s2n.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+#define STRING_LEN 1024
+static char str_buffer[STRING_LEN];
+
+#define REPLY(...) sprintf(str_buffer, __VA_ARGS__);\
+    s2n_send(conn, str_buffer, strlen(str_buffer), &blocked);
+
+/*
+ * simple https handler that allows https clients to connect
+ * but currently does not do any user parsing
+ */
+int https(struct s2n_connection *conn, bool bench)
+{
+    const char header[] = "HTTP/1.1 200 OK\r\n\r\n";
```

Comment:
Since you are neither setting a content-length nor do chunked encoding the client might not really interpret your end of the response correctly.

maybe rather build a big blob with all the info you want to send, set the content-length and then send it.
You can also set a big content-length and then kill the request after having send all you need, but some clients will interpret that as an error.

Reply:
Yes. But I think most clients of today will do HTTP/1.1 😀

Reply:
yeah right this closes the connection after sending the body. most clients can handle this (just like most browsers can deal with unclosed tags)

Reply:
Yeah, but you can reply to HTTP/1.1 request with HTTP/1.0 response. That said, I'd rather prefer HTTP/1.1 Content-Length here (:

Reply:
we don't really know whether we are responding to which requests since this simple server doesn't parse inputs, but i can add support for Content-Length.